src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6054,7 +6053,7 @@ private BoundExpression BindArrayAccess(ExpressionSyntax node, BoundExpression e
                 }
             }
 
-            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), arrayType.ElementType);
+            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), arrayType.ElementType, hasErrors);
```

Comment:
Not really. The reported diagnostic should prevent downstream processing anyway. I'm just preserving existing behavior while adding this error.

---
In reply to: [75064678](https://github.com/dotnet/roslyn/pull/13091#discussion_r75064678) [](ancestors = 75064678)